Changes to regenerators and healing effects

CrabbytheCrab opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Ports https://github.com/Lobotomy-Corporation-13/lc13/pull/182, https://github.com/Lobotomy-Corporation-13/lc13/pull/218/, and https://github.com/Lobotomy-Corporation-13/lc13/pull/251, all by Aseidos
It ports a refractor to regenerators and an system for indexing different lists of mobs in an area which removes mobs that died from it. Most hostile mobs that were passive to humans won't be able to trigger the regenerators reduced healing anymore. I also added a variable for abnormalities called breach_index to make changing an abnormality's index when they breach less hard coded and a MOB_ABNO_PASSIVE_INDEX which is used by abnormalities that aren't hostile during their breach or are currently contained.
This PR also does some other minor changes and balancing. Burst mode now works with the regenerator lob upgrade and will display changes to its healing and the amount of healing it gives when it goes off. Refined Enkephalin and Mental Stabilizers now heal half the amount of sanity and the former will now cause hallucinations when a player overdoses on it. When a human heals hp, They will spawn a healing flick effect and also redid it for sanity healing as well.
Also updates the description of "A prayer to god" to better reflect its effects
## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
The ports not only reduces lag but also allows for area wide effects.
Refined Enkephalin and Mental Stabilizers were never nerfed from the pre hp rework pr and the former didn't do anything on overdose despite mentioning causing hallucinations .
Standardizing healing particles makes them look better.

## Changelog
:cl:
balance: rebalanced Refined Enkephalin and Mental Stabilizer
icons: added some new ones for sanity healing
code: Ported a refactor to regenerators

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

CrabbytheCrab committed:
finished fixes

CrabbytheCrab committed:
no sanity drain

CrabbytheCrab committed:
Update medicine_reagents.dm

CrabbytheCrab committed:
Update abnormality_queue.dm

CrabbytheCrab committed:
ports prs

CrabbytheCrab committed:
Update areas.dm

CrabbytheCrab committed:
added healing effects to regenerators

CrabbytheCrab committed:
healing particles

CrabbytheCrab committed:
Removed healing effects from abilities and weapons

CrabbytheCrab committed:
changes undos and ports

CrabbytheCrab committed:
icon changes and fixes

CrabbytheCrab committed:
refractor stuff

CrabbytheCrab committed:
more changes

CrabbytheCrab committed:
moth

CrabbytheCrab committed:
fixes

CrabbytheCrab committed:
Update 32x32.dmi

CrabbytheCrab committed:
updated the effects to use the flick system

CrabbytheCrab committed:
undoes the regenerator nerfs

Coxswain-Navigator reviewed (changes requested):
Some small nitpicks. Ensure this is tested locally before testmerging.

CrabbytheCrab committed:
nitpick fixes

CrabbytheCrab committed:
Minor changes

Update puss_in_boots.dm

Update puss_in_boots.dm

Update wrath_servant.dm

Update puss_in_boots.dm

Update wrath_servant.dm

CrabbytheCrab committed:
Removed Healing Effects on regenerators

CrabbytheCrab committed:
removed area stuff that I left in by accident

CrabbytheCrab committed:
Update lobotomy_corp.dm

Coxswain-Navigator reviewed (approved):
We've tested multiple iterations of this PR by now, and I think it's about time to merge.
